modules/rpc/handler.go

```diff
@@ -97,3 +98,41 @@ func (f fxContextOnewayInboundMiddleware) HandleOneway(ctx context.Context, req
 	fxctx := fxcontext.New(ctx, f.Host)
 	return handler.HandleOneway(fxctx, req)
 }
+
+type authInboundMiddleware struct {
+	service.Host
+}
+
+func (a authInboundMiddleware) Handle(ctx context.Context, req *transport.Request, resw transport.ResponseWriter, handler transport.UnaryHandler) error {
+	fxctx, err := authorize(ctx, a.Host)
+	if err != nil {
+		return err
```

Comment:
Interesting. BadRequestError is an internal package to YARPC (with the version I have with fx, before /api/transport) I will add a TODO once upgrade happens, to move the error to BadRequestError.